game/scripts/vscripts/components/cave/cave.lua

```diff
@@ -171,11 +171,26 @@ function CaveHandler:CreepDeath (teamID, roomID)
       cave.rooms[roomID + 1].door.Open()
       cave.rooms[roomID + 1].zone.disable()
 
-      -- inform players
-      Notifications:TopToTeam(teamID, {
-        text = "Room " .. roomID .. " got cleared. You can now advance to the next room",
-        duration = 5,
-      })
+      local result = FindUnitsInRadius(
+        teamID, -- team
+        cave.rooms[roomID].zone.origin, -- location
+        nil, -- cache
+        cave.rooms[roomID].radius, -- radius
+        DOTA_UNIT_TARGET_TEAM_FRIENDLY, -- team filter
+        DOTA_UNIT_TARGET_ALL, -- type filter
+        DOTA_UNIT_TARGET_FLAG_NONE, -- flag filter
+        FIND_ANY_ORDER, -- order
+        false -- can grow cache
+      )
+      for _, unit in pairs(result) do
+        if CaveHandler:IsInFarmingCave(teamID, unit) then
+          -- inform players
+          Notifications:Top(unit:GetPlayerOwner(), {
```

Comment:
This will go off for every unit a player has in the farming cave. You should make sure to only send the notification once per player ID. simple map and check would work, `hasSeenNotification[playerID] = true`